rotkehlchen/accounting/cost_basis/base.py

```diff
@@ -256,34 +315,21 @@ def reduce_asset_amount(self, asset: Asset, amount: FVal, timestamp: Timestamp)
         This function does the same as calculate_spend_cost_basis as far as consuming
         acquisitions is concerned but does not calculate bought cost.
         """
-        # No need to do anything if amount is to be reduced by zero
-        if amount == ZERO:
```

Comment:
I've removed several checks like `amount == ZERO` and `len(asset_events.acquisitions) == 0` because they are not needed and the algorithm handles these cases properly